src/hardware/serialport/softmodem.cpp

```diff
@@ -258,31 +272,31 @@ void CSerialModem::Reset(){
 void CSerialModem::EnterIdleState(void){
 	connected=false;
 	ringing=false;
-	
-	if(clientsocket) {
-		delete clientsocket;
-		clientsocket=0;
-	}
+	clientsocket.reset(nullptr);
+	waitingclientsocket.reset(nullptr);
 
-	if(waitingclientsocket) {	// clear current incoming socket
-		delete waitingclientsocket;
-		waitingclientsocket=0;
-	}
 	// get rid of everything
 	if(serversocket) {
-		while( (waitingclientsocket=serversocket->Accept()) )
-			delete waitingclientsocket;
+		waitingclientsocket.reset(serversocket->Accept());
+		while (waitingclientsocket) {
+			waitingclientsocket.reset(serversocket->Accept());
+		}
 	} else if (listenport) {
-		
-		serversocket=new TCPServerSocket(listenport);	
+
+		serversocket.reset(new TCPServerSocket(listenport));
 		if(!serversocket->isopen) {
-			LOG_MSG("Serial%d: Modem could not open TCP port %d.",COMNUMBER,listenport);
-			delete serversocket;
-			serversocket=0;
-		} else LOG_MSG("Serial%d: Modem listening on port %d...",COMNUMBER,listenport);
+			LOG_MSG("Serial%u: Modem could not open TCP port %u.",
+			        static_cast<uint32_t>(COMNUMBER),
+			        static_cast<uint32_t>(listenport));
+
+			serversocket.reset(nullptr);
+		} else
+			LOG_MSG("Serial%u: Modem listening on port %u...",
+		            static_cast<uint32_t>(COMNUMBER),
```

Comment:
Some spaces want to be replaced by a tab in here.

Reply:
Will do.

Reply:
Fixed in f5319290